lib/hostip.c

```diff
@@ -659,6 +659,13 @@ enum resolve_t Curl_resolv(struct Curl_easy *data,
   (void)allowDOH;
 #endif
 
+  /* We should intentionally error and not resolve .onion TLDs */
+  char *last_dot = strrchr(hostname, '.');
+  if(last_dot && strcmp(".onion", last_dot) == 0) {
+    infof(data, "Not resolving .onion address (RFC 7686)");
```

Comment:
1. This should probably be failf() instead of infof().
2. We usually use `!strcmp()...` instead of comparing with `== 0`.
3. Shouldn't the check be case insensitive? Host names are.